nodejs/aws-infra/experimental/clusterAutoScaling.ts

```diff
@@ -0,0 +1,579 @@
+        repo_upgrade_exclude:
+            - kernel*
+        packages:
+            - aws-cfn-bootstrap
+            - aws-cli
+            - nfs-utils
+        mounts:
+            - ['/dev/xvdb', 'none', 'swap', 'sw', '0', '0']
+        bootcmd:
+            - mkswap /dev/xvdb
+            - swapon /dev/xvdb
+            - echo ECS_CLUSTER='${clusterId}' >> /etc/ecs/ecs.config
+            - echo ECS_ENGINE_AUTH_TYPE=docker >> /etc/ecs/ecs.config
+        runcmd:
+            # Set and use variables in the same command, since it's not obvious if
+            # different commands will run in different shells.
+            - |
+                # Knock one letter off of availability zone to get region.
+                AWS_AVAILABILITY_ZONE=$(curl -s 169.254.169.254/2016-09-02/meta-data/placement/availability-zone)
+                AWS_REGION=$(echo $AWS_AVAILABILITY_ZONE | sed 's/.$//')
+
+                ${fileSystemRuncmdBlock}
+
+                # Disable container access to EC2 metadata instance
+                # See http://docs.aws.amazon.com/AmazonECS/latest/developerguide/instance_IAM_role.html
+                iptables --insert FORWARD 1 --in-interface docker+ --destination 169.254.169.254/32 --jump DROP
+                service iptables save
+
+                /opt/aws/bin/cfn-signal \
+                    --region "\${AWS_REGION}" \
+                    --stack "${cloudFormationStackName}" \
+                    --resource Instances
+        `;
+    });
+}
+
+export class ClusterAutoScalingGroup extends pulumi.ComponentResource {
+    public readonly instance: aws.cloudformation.Stack;
+
+    public readonly cluster: mod.Cluster;
+
+    /**
+     * The launch configuration for this auto scaling group.
+     */
+    public readonly launchConfiguration: ClusterAutoScalingLaunchConfiguration;
+
+    constructor(name: string,
+                args: ClusterAutoScalingGroupArgs,
+                opts: pulumi.ComponentResourceOptions = {}) {
+        super("aws-infra:x:ClusterAutoScalingGroup", name, args, opts);
+
+        const parentOpts = { parent: this };
+        const cluster = args.cluster;
+
+        let launchConfiguration: ClusterAutoScalingLaunchConfiguration;
+
+        // Use the autoscaling config provided, otherwise just create a default one for this cluster.
+        if (args.launchConfiguration) {
+            launchConfiguration = args.launchConfiguration;
+        }
+        else {
+            launchConfiguration = new ClusterAutoScalingLaunchConfiguration(
+                name, args.launchConfigurationArgs || { cluster }, parentOpts);
+        }
+
+        this.instance = new aws.cloudformation.Stack(name, {
+            ...args,
+            name: launchConfiguration.stackName,
+            templateBody: getCloudFormationTemplate(
+                name,
+                launchConfiguration.instance.id,
+                cluster.network.subnetIds,
+                args.templateParameters || {}),
```

Comment:
find terraform's autoscaling group args bag and use that.  aws.autoscaling.group.